src/main/java/org/reactome/release/qa/check/MissingEditorialAttributeCheck.java

```diff
@@ -0,0 +1,99 @@
+package org.reactome.release.qa.check;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.schema.GKSchemaClass;
+import org.gk.schema.InvalidAttributeException;
+import org.reactome.release.qa.annotations.SliceQACheck;
+import org.reactome.release.qa.common.AbstractQACheck;
+import org.reactome.release.qa.common.QAReport;
+
+/**
+ * This QA check was requested by Lisa.
+ * Description:
+ * "Weekly QA and release QA: Report new events in
+ * slice (those where stableID released attribute
+ * is != True) and edited, reviewed, or authored
+ * attribute is null."
+ * @author sshorser
+ *
+ */
+@SliceQACheck
+public class MissingEditorialAttributeCheck extends AbstractQACheck {
+
+	private static final String YES = "YES";
+	private static final String NO = "NO";
+	private static Set<String> reportedEntities = new HashSet<>();
+
+	@Override
+	public QAReport executeQACheck() throws Exception
+	{
+		GKSchemaClass eventClass = (GKSchemaClass) this.dba.getSchema().getClassByName(ReactomeJavaConstants.Event);
+		QAReport report = new QAReport();
+		report.setColumnHeaders("DB_ID", "Name", "Edited?", "Authored?", "Reviewed?", "Created");
+		Collection<GKSchemaClass> classesToCheck = (Collection<GKSchemaClass>) eventClass.getSubClasses();
+		// make sure Event is in there with its subclasses.
+		classesToCheck = new ArrayList<>(classesToCheck);
+		classesToCheck.add(eventClass);
+		for (GKSchemaClass schemaClass : classesToCheck)
+		{
+			Collection<GKInstance> eventInstances = (Collection<GKInstance>) this.dba.fetchInstancesByClass(schemaClass);
+			for (GKInstance eventInstance : eventInstances)
+			{
+				String[] line = getReportLine(eventInstance);
+				if (line.length > 0)
+				{
+					report.addLine(line);
+				}
+			}
+		}
+		return report;
+	}
+
+	/**
+	 * Returns an array of Strings that will be added as a line to the report - IFF the eventInstance meets the criteria:
+	 * StableIdentifier is NOT released; Any of: authored, edited, reviewed are null.
+	 * @param eventInstance
+	 * @return
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private String[] getReportLine(GKInstance eventInstance) throws InvalidAttributeException, Exception
+	{
+		ArrayList<String> line = new ArrayList<>();
+		GKInstance stableIdentifier = (GKInstance) eventInstance.getAttributeValue(ReactomeJavaConstants.stableIdentifier);
+		Boolean released = (Boolean) stableIdentifier.getAttributeValue(ReactomeJavaConstants.released);
```

Comment:
Since the value in "released" is stored as a String of "True" or "False" (with potentially varying case), it might be better to use `Boolean.parseBoolean(String)`:

`boolean released = Boolean.parseBoolean(stableIdentifier.getAttributeValue(ReactomeJavaConstants.released)`

This will also let you skip the null check below as passing a null value to the parseBoolean method will return false which fits the logic well.

Reply:
Are you sure that won't just _move_ the null check to a new location?

Reply:
I'm sure - if the value returned from `stableIdentifier.getAttributeValue(ReactomeJavaConstants.released)` is null, then `Boolean.parseBoolean` will return false.  As long as it's stored in a primitive boolean type, you don't need the null check further down.

Reply:
This:
```Java
boolean released = Boolean.parseBoolean( (String) stableIdentifier.getAttributeValue(ReactomeJavaConstants.released));
``` 
causes a class cast exception: 
> Exception in thread "main" java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
>	at org.reactome.release.qa.check.MissingEditorialAttributeCheck.getReportLine(MissingEditorialAttributeCheck.java:76)
>	at org.reactome.release.qa.check.MissingEditorialAttributeCheck.executeQACheck(MissingEditorialAttributeCheck.java:48)
>	at org.reactome.release.qa.Main.main(Main.java:140)

That's probably why I did it the other way. I think getAttributeValue is returning a `Boolean` in this case, not a String.

Reply:
Ah okay - I thought that a String was returned.  In that case, the way you have it makes sense.